File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/impl/editor/codestyle/java/JavaCodeStyleSettingsApplier.java
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.codestyle.java;

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.JavaCodeStyleSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.arrangement.JavaArrangementSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.blank_lines.JavaBlankLinesSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.imports.JavaImportsSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc.JavadocSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces.JavaWrappingAndBracesSettings;
import javax.inject.Inject;

public class JavaCodeStyleSettingsApplier implements SettingsApplier<JavaCodeStyleSettings> {

  private final SettingsApplier<JavaArrangementSettings> javaArrangementSettingsApplier;
  private final SettingsApplier<JavaBlankLinesSettings> javaBlankLinesSettingsApplier;
  private final SettingsApplier<JavaImportsSettings> javaImportsSettingsApplier;
  private final SettingsApplier<JavadocSettings> javadocSettingsApplier;
  private final SettingsApplier<JavaWrappingAndBracesSettings> javaWrappingAndBracesSettingsApplier;

  @Inject
  public JavaCodeStyleSettingsApplier(SettingsApplier<JavaArrangementSettings> javaArrangementSettingsApplier,
                                      SettingsApplier<JavaBlankLinesSettings> javaBlankLinesSettingsApplier,
                                      SettingsApplier<JavaImportsSettings> javaImportsSettingsApplier,
                                      SettingsApplier<JavadocSettings> javadocSettingsApplier,
                                      SettingsApplier<JavaWrappingAndBracesSettings> javaWrappingAndBracesSettingsApplier) {
    this.javaArrangementSettingsApplier = javaArrangementSettingsApplier;
    this.javaBlankLinesSettingsApplier = javaBlankLinesSettingsApplier;
    this.javaImportsSettingsApplier = javaImportsSettingsApplier;
    this.javadocSettingsApplier = javadocSettingsApplier;
    this.javaWrappingAndBracesSettingsApplier = javaWrappingAndBracesSettingsApplier;
  }

  public void apply(JavaCodeStyleSettings settings) {
    settings.arrangement().ifPresent(javaArrangementSettingsApplier::apply);
    settings.blankLines().ifPresent(javaBlankLinesSettingsApplier::apply);
    settings.imports().ifPresent(javaImportsSettingsApplier::apply);
    settings.javadoc().ifPresent(javadocSettingsApplier::apply);
    settings.wrappingAndBraces().ifPresent(javaWrappingAndBracesSettingsApplier::apply);
  }
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/applier/impl/editor/codestyle/java/JavaCodeStyleSettingsApplier.kt
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.codestyle.java

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.JavaCodeStyleSettings
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.arrangement.JavaArrangementSettings
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.blank_lines.JavaBlankLinesSettings
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.imports.JavaImportsSettings
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc.JavadocSettings
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces.JavaWrappingAndBracesSettings
import javax.inject.Inject

class JavaCodeStyleSettingsApplier @Inject
constructor(private val javaArrangementSettingsApplier: SettingsApplier<JavaArrangementSettings>,
            private val javaBlankLinesSettingsApplier: SettingsApplier<JavaBlankLinesSettings>,
            private val javaImportsSettingsApplier: SettingsApplier<JavaImportsSettings>,
            private val javadocSettingsApplier: SettingsApplier<JavadocSettings>,
            private val javaWrappingAndBracesSettingsApplier: SettingsApplier<JavaWrappingAndBracesSettings>) : SettingsApplier<JavaCodeStyleSettings> {

    override fun apply(settings: JavaCodeStyleSettings) {
        settings.arrangement?.let(javaArrangementSettingsApplier::apply)
        settings.blankLines?.let(javaBlankLinesSettingsApplier::apply)
        settings.imports?.let(javaImportsSettingsApplier::apply)
        settings.javadoc?.let(javadocSettingsApplier::apply)
        settings.wrappingAndBraces?.let(javaWrappingAndBracesSettingsApplier::apply)
    }
}
